src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java.g

```diff
@@ -207,6 +208,15 @@ tokens {
     {
         return ((currentLtLevel != 0) || ltCounter == currentLtLevel);
     }
+
+    /**
+    * This int value tracks the depth of a switch expression. Along with the
+    * IDENT to id rule at the end of the parser, this value helps us
+    * to know if the "yield" we are parsing is an IDENT, method call, class,
+    * field, etc. or if it is a java 13+ yield statement. Positive values
+    * indicate that we are within a (possibly nested) switch expression.
+    */
+    private int switchRuleOrStatementDepth = 0;
```

Comment:
Would a name like "switchBlockDepth" make sense here?

Reply:
Yes, that does make more sense.

Reply:
Fixed.